Hooks library, Various Enhancements & Plugin example using hooks

echiu64 opened the pull request:
Added in hooks library (ZPL). This should make core functionality more core and more functionality can be pushed to plugins via hooks. A few hooks have been added so far. More hooks to be added in the future.
Add example plugin that shows how hooks can be used
Wrapped couple of traceback calls if debug is enabled
Removed bytereverse in LP (wordreverse is good enough)
Removed check for negative LP penalties in web

echiu64 committed:
New plugin hooks library (ZPL)

echiu64 committed:
Initial hooks for LP class

echiu64 committed:
Allow for negative penalties

echiu64 committed:
Various minor fixes in LP, added back traceback if debugging enabled

echiu64 committed:
Traceback if debug enabled (work.py)

echiu64 committed:
Example plugin using hooks

echiu64 committed:
Merge branch 'master' of https://github.com/c00w/bitHopper into 247

c00w commented:
Please add the peak license to a file called peak_LICENSE and release your work into the public domain. Also there is some weird indentation in lp.py. You should probably delete lp_callback.py since that was a worse implementation of the exact same thing. I can do most of this if you're too busy.

echiu64 commented:
I can do it, my change, my responsibility ;)
Except for removing lp_callback.py, didn't even notice it was there, would rather if someone more familiar with that remove it

echiu64 committed:
Add license file for Peak libraries

echiu64 committed:
Fix indentation issues

echiu64 commented:
Ok, done, should be ready for use.

Added 2 commits: added license file for Peak libraries and fixed the indentation for lp.py, (thought I fixed Komodo to use spaces instead of tabs)
